### ISSUE TITLE
Creating a report for a workflow with script, wrapper, and notebook directives fails

### ISSUE THREAD

**Issue by HenningTimm** [snakemake/snakemake#745]
**Snakemake version**
5.28.0

**Describe the bug**
Running `snakemake --report` for a workflow containing rules with script, wrapper, and notebook as well as a report fails with `NameError`.

**Logs**
```
Building DAG of jobs...
Creating report...
Adding dummy_result.txt (1.7e-05 MB).
Downloading resources and rendering HTML.
Traceback (most recent call last):
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/__init__.py", line 687, in snakemake
    success = workflow.execute(
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/workflow.py", line 820, in execute
    auto_report(dag, report, stylesheet=report_stylesheet)
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/report/__init__.py", line 880, in auto_report
    rendered = template.render(
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/report/report.html.jinja2", line 538, in top-level template code
    {% if rule.code %}
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/jinja2/environment.py", line 471, in getattr
    return getattr(obj, attribute)
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/common.py", line 136, in __get__
    value = self.method(instance)
  File "/home/me/miniconda3/envs/bleeding_edge_snakemake/lib/python3.9/site-packages/snakemake/report/__init__.py", line 260, in code
    elif self._rule.script is not None and not contains_wildcard(self._rule.script):
NameError: name 'contains_wildcard' is not defined
```

**Minimal example**

```
# Snakefile
rule dummy:
    output:
        report("dummy_result.txt")
    script:
        "test.py"

rule all:
    "dummy_result.txt"
```

```
# test.py
with open(snakemake.output[0], "w") as out_file:
    print("this is a result", file=out_file)
```

Call with:
```
snakemake
snakemake --report
```

**Additional context**
This is due to a missing import. I will submit a PR to fix this.

**Comment by cpauvert** [comment]
Thanks @HenningTimm for spotting the incriminated lines and suggesting a PR. I have the same issue!

**Comment by johanneskoester** [comment]
Closed by pr #746

**Comment by johanneskoester** [comment]
thanks!
